### PR TITLE
Private events

### PR DESCRIPTION
**Pull request by vikitaotiz**
The project involves the creation of a private event hosting platform.
I have created a user model and authentication using devise gem.
I have created an event model and added a relationship to a user.
Events are divided into past and upcoming.
Events have many attendees

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+development:
+  <<: *default
+  database: db/development.sqlite3
+
+# Warning: The database defined as "test" will be erased and
+# re-generated from your development database when you run "rake".
+# Do not set this db to the same as development or production.
+test:
+  <<: *default
+  database: db/test.sqlite3
```
Comment (eliasecasta) [inline config/database.yml]: Please migrate your app to work 100% with PostgreSQL, you cant use SQLite in this project at all for practice motives.

File: config/database.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+production:
+  adapter: postgresql
+  username: postgres
+  password: root
```
Comment (eliasecasta) [inline config/database.yml]: Remove your DB credentials so they don't get exposed to the public.

File: Gemfile [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+gem 'bootsnap', '>= 1.4.4', require: false
+
+group :production do
+  gem 'pg'
+end
+
+group :development, :test do
+  # Use sqlite3 as the database for Active Record
+  gem 'sqlite3', '~> 1.4'
```
Comment (eliasecasta) [inline Gemfile]: Do not use sqlite anywhere

**Review by eliasecasta — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️

Hello  @vikitaotiz,

Good job so far with your project, I will mention mandatory things that you need to fix before moving on:

- [ ] Get rid of all SQLite usage, you should use only PostgreSQL in this project.
- [ ] A not logged in user should not be able to see the Attend event button, your app crashes if I click it while not logged in.
- [ ] The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [ ] Navigation links are present at the top of the page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/navigation_links_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)) (Add navigations for the user's show page)

#### Associations
##### User as event creator

- [ ] User model: `has_many :events, foreign_key: :creator_id` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

- [ ] Event model: `belongs_to :creator, class_name: "User"` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

##### Migrations
- [ ] Events table has creator_id as the foreign key ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))


```ruby
def change
  create_table :users do |t|
  end

  create_table :events do |t|
    t.integer :creator_id
  end
end
```

- [ ] the events table has column `creator_id` of type `integer`
- [ ] an index for `creator_id` column is added

```markdown
Why are indices for the foreign keys necessary? In short:
Since the foreign key columns will be used often to retrieve related information, It's important to index them (queries on those columns will be faster with indexes).
More details:

- [Helpful Link One](https://semaphoreci.com/blog/2017/05/09/faster-rails-is-your-database-properly-indexed.html)
- [Helpful Link Two](https://sqlperformance.com/2012/11/t-sql-queries/benefits-indexing-foreign-keys).
```

#### Tests

- [ ] Add Rspec gem to your Gemfile
- [ ] You didn't add any test, please create unit and integration tests:

- [ ] Models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

- [ ] Authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

- [ ] Event management features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

Please correct those issues and send another review request, good luck, and well done so far!

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# STATUS: Changes Required :red_circle: 

## Hello @vikitaotiz, you've done a great job so far, thank you for implementing almost all the requirements asked by the previous TSE. However, you still need to improve the following which was asked previously:

- [ ] Add more unit and integration tests. Keep in mind to  properly configure your RSpec and make sure all of of your tests are passing
- [ OPTIONAL ] I'm adding this as optional because it was not mentioned by the previous TSE but highly recommended. Please explain in your readme file how to install RSpec and how to test your project with rspec 

### If you have any questions with regards to the requirements please add a comment here and I will try to reply ASAP.

# HAPPY CODING :computer: :desktop_computer:

**Review by carlos-ssh — APPROVED** [review]
#  🍾🍾🍾 STATUS: APPROVED 🍾🍾🍾
#
Hello Team 👋🏽👋🏽👋🏽
This is the result of your hard work.
Congratulations guys. 👏🏽
You can merge and continue with your path.

Happy Coding! 👾
@Carlos-ssh

**Pull request merged**